python_ta/reporters/node_printers.py

```diff
@@ -614,6 +614,36 @@ def render_pep8_errors_e306(msg, _node, source_lines=None):
     yield from render_context(msg.line, msg.line + 2, source_lines)
 
 
+def render_missing_return_statement(msg, node, source_lines=None):
+    """
+    Render a missing return statements message
+    """
+    yield from render_context(msg.end_line - 1, msg.end_line + 1, source_lines)
+
+    # calculate indentation for the insertion point
+    body = source_lines[msg.end_line - 1]
+    indentation = len(source_lines[msg.line - 1]) - len(source_lines[msg.line - 1].lstrip())
+
+    # determine whether reaching the end of function
+    function_indentation = len(source_lines[node.lineno - 1]) - len(
+        source_lines[node.lineno - 1].lstrip()
+    )
+    if msg.end_line == node.end_lineno and indentation == function_indentation + 4:
```

Comment:
This is quite good, but you shouldn't assume that indentation is always 4 spaces. A more robust approach would be to compare this indentation against the indentation of the first statement in the function body